File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Data.Filtering;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid;
using System.Collections;

namespace WarnNewRowDisappears {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            new DevExpress.XtraGrid.Design.XViewsPrinting(gridControl1);
            gridView1.OptionsView.ShowGroupPanel = false;
            gridView1.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom;
            gridView1.ActiveFilterCriteria = new OperandProperty("Discontinued") == new OperandValue(false);
        }

        bool IsRowFilteredOut(GridView view, object dataRow) {
            if(view == null || view.DataSource == null || dataRow == null)
                throw new ArgumentNullException();
            int dataSourceIndex = ((IList)view.DataSource).IndexOf(dataRow);
            if(dataSourceIndex < 0) 
                throw new Exception("Row doesn't belong to the data source");
            return view.GetRowHandle(dataSourceIndex) < 0;
        }

        private void gridView1_RowUpdated(object sender, DevExpress.XtraGrid.Views.Base.RowObjectEventArgs e) {
            GridView view = (GridView)sender;
            if(view.ActiveFilterEnabled && view.FocusedRowHandle == GridControl.NewItemRowHandle) {
                if(IsRowFilteredOut(view, e.Row))
                    MessageBox.Show("The new row you've just inserted is filtered out of the view due to the active filter.");
            }
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Data.Filtering
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid
Imports System.Collections

Namespace WarnNewRowDisappears
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim TempXViewsPrinting As DevExpress.XtraGrid.Design.XViewsPrinting = New DevExpress.XtraGrid.Design.XViewsPrinting(gridControl1)
			gridView1.OptionsView.ShowGroupPanel = False
			gridView1.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom
			gridView1.ActiveFilterCriteria = New OperandProperty("Discontinued") = New OperandValue(False)
		End Sub

		Private Function IsRowFilteredOut(ByVal view As GridView, ByVal dataRow As Object) As Boolean
			If view Is Nothing OrElse view.DataSource Is Nothing OrElse dataRow Is Nothing Then
				Throw New ArgumentNullException()
			End If
			Dim dataSourceIndex As Integer = (CType(view.DataSource, IList)).IndexOf(dataRow)
			If dataSourceIndex < 0 Then
				Throw New Exception("Row doesn't belong to the data source")
			End If
			Return view.GetRowHandle(dataSourceIndex) < 0
		End Function

		Private Sub gridView1_RowUpdated(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.RowObjectEventArgs) Handles gridView1.RowUpdated
			Dim view As GridView = CType(sender, GridView)
			If view.ActiveFilterEnabled AndAlso view.FocusedRowHandle = GridControl.NewItemRowHandle Then
				If IsRowFilteredOut(view, e.Row) Then
					MessageBox.Show("The new row you've just inserted is filtered out of the view due to the active filter.")
				End If
			End If
		End Sub
	End Class
End Namespace